app/src/marketplace/components/StreamPreviewPage/index.jsx

```diff
@@ -104,6 +106,16 @@ const StreamPreviewPage = ({
         setHasData(true)
     }, [isMounted])
 
+    const permissions = useStreamPermissions(urlId)
+
+    if (!permissions && !productId) {
```

Comment:
If `productId` is defined then we go with the old approach = no permissions check.